svc.go

```diff
@@ -167,21 +171,83 @@ func (as *autoNATService) doDial(pi peer.AddrInfo) *pb.Message_DialResponse {
 	ctx, cancel := context.WithTimeout(as.ctx, as.config.dialTimeout)
 	defer cancel()
 
-	as.config.dialer.Peerstore().ClearAddrs(pi.ID)
+	// START DIALING
```

Comment:
This logic, while i agree this is a logical approach from the service side, does make our overall reasoning really tricky, and is why we were cautious about doing something like this before:
1. The client who's asking for dial backs is going to expect that any individual auto nat service isn't going to dial it back on all of it's potential candidate addresses. However, it is also the entity in the position of knowing which addresses it has more confidence on being public/private, and which ones haven't been tested yet.
2. That means it has more knowledge. and need to prioritize which addresses should be tested. Having there be additional selection logic opaque to the client on which subset of candidates it sends to the service for testing makes the client's life much harder.
3. as such, it's worth considering: can we have this selection logic happen on the client, where it only sends the candidates that make it through this selection process to the server, and the server tests all of them.

Reply:
Early on, the autonat service will have more information because it'll be able to see external addresses.

However, I agree in general. We should test addresses in an order specified by the client.

But there may be some room to mix in some additional address guesses. (maybe let the client specify a guess true/false in the request)